website/project/decorators.py

```diff
@@ -13,33 +13,40 @@
 
 from website.models import Node
 
+def _load_node_or_fail(nid):
```

Comment:
I think it would be useful to have a general purpose `get_or_404(Model, pk)` function. Perhaps put it in `website.utils` or `framework.mongo` modules.